thevoid/connection.cpp

```diff
@@ -19,6 +19,9 @@
 #include <boost/bind.hpp>
 #include <iostream>
 
+#include <boost/spirit/include/karma.hpp>
+#include <boost/fusion/include/std_pair.hpp>
```

Comment:
Let me clarify this - you want to add grammar generators just to print several headers into log?
What about embedding LISP interpreter to find out needed headers?

Did you ever run performance test with this implementation, basic printf and without header dump?